### PR TITLE
Update Msal

### PR DESCRIPTION
**Pull request by neha-bhargava**

### CODE REVIEW

File: TodoListClient/TodoListClient.csproj [hunk 1]
```diff
@@ -15,6 +15,8 @@
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <Deterministic>true</Deterministic>
     <TargetFrameworkProfile />
+    <NuGetPackageImportStamp>
```
Comment (bgavrilMS) [inline TodoListClient/TodoListClient.csproj:18]: pls delete empty properties

File: TodoListService/TodoListService.csproj [hunk 2]
```diff
@@ -337,8 +340,15 @@
     <Error Condition="!Exists('..\packages\EntityFramework.6.3.0\build\EntityFramework.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\EntityFramework.6.3.0\build\EntityFramework.props'))" />
     <Error Condition="!Exists('..\packages\EntityFramework.6.3.0\build\EntityFramework.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\EntityFramework.6.3.0\build\EntityFramework.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.1\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.1\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props'))" />
+    <Error Condition="!Exists('..\packages\System.Runtime.WindowsRuntime.4.6.0\build\net461\System.Runtime.WindowsRuntime.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\System.Runtime.WindowsRuntime.4.6.0\build\net461\System.Runtime.WindowsRuntime.targets'))" />
```
Comment (bgavrilMS) [inline TodoListService/TodoListService.csproj:343]: Are these needed?

File: TodoListService/TodoListService.csproj [hunk 3]
```diff
@@ -337,8 +340,15 @@
...
+    <Error Condition="!Exists('..\packages\Microsoft.Windows.SDK.Contracts.10.0.17763.1000\build\Microsoft.Windows.SDK.Contracts.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.SDK.Contracts.10.0.17763.1000\build\Microsoft.Windows.SDK.Contracts.props'))" />
+    <Error Condition="!Exists('..\packages\Microsoft.Windows.SDK.Contracts.10.0.17763.1000\build\Microsoft.Windows.SDK.Contracts.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.SDK.Contracts.10.0.17763.1000\build\Microsoft.Windows.SDK.Contracts.targets'))" />
   </Target>
   <Import Project="..\packages\EntityFramework.6.3.0\build\EntityFramework.targets" Condition="Exists('..\packages\EntityFramework.6.3.0\build\EntityFramework.targets')" />
+  <Import Project="..\packages\System.Runtime.WindowsRuntime.4.6.0\build\net461\System.Runtime.WindowsRuntime.targets" Condition="Exists('..\packages\System.Runtime.WindowsRuntime.4.6.0\build\net461\System.Runtime.WindowsRuntime.targets')" />
```
Comment (bgavrilMS) [inline TodoListService/TodoListService.csproj:349]: Are this still needed? I think not.

File: TodoListService/packages.config [hunk 4]
```diff
@@ -16,7 +16,6 @@
   <package id="Microsoft.AspNet.WebApi.WebHost" version="5.2.7" targetFramework="net472" />
   <package id="Microsoft.AspNet.WebPages" version="3.2.7" targetFramework="net472" />
   <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="2.0.1" targetFramework="net472" />
-  <package id="Microsoft.Identity.Client" version="4.22.0" targetFramework="net472" />
```
Comment (bgavrilMS) [inline TodoListService/packages.config:19]: Hmm, interesting - so you now have some references via packages.config (legacy) and some references via packagereference. Ideally they should all be in the csproj and you can delete packges.config

**Comment by neha-bhargava** [comment]
Reverted the last update and updated again with the trick @bgavrilMS suggested. https://github.com/NuGet/docs.microsoft.com-nuget/issues/860#issuecomment-409207305

File: TodoListService/TodoListService.csproj [hunk 5]
```diff
@@ -298,9 +191,62 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
-    <None Include="packages.config" />
     <Content Include="Scripts\jquery-3.4.1.slim.min.map" />
     <Content Include="Scripts\jquery-3.4.1.min.map" />
+    <None Include="app.config" />
+  </ItemGroup>
+  <ItemGroup>
```
Comment (bgavrilMS) [inline TodoListService/TodoListService.csproj:198]: Nice. So the trick worked?

**Review by trwalke — APPROVED** [review]
Looks good. Thanks Neha

File: TodoListService/TodoListService.csproj (hunk 5 above)
Comment (neha-bhargava) [reply TodoListService/TodoListService.csproj:198]: Yes it did. But after applying the trick I had to add back a lot of stuff it removed and change. But a lot less work than manually doing it. :)

**Pull request merged**